core/src/main/java/nl/nn/adapterframework/compression/ZipWriter.java

```diff
@@ -15,284 +15,103 @@
+import java.util.List;
 import java.util.zip.ZipOutputStream;
 
-import javax.servlet.http.HttpServletResponse;
-
 import org.apache.commons.lang3.StringUtils;
-import org.apache.logging.log4j.Logger;
+import org.springframework.http.MediaType;
 
-import lombok.Getter;
-import lombok.Setter;
-import nl.nn.adapterframework.collection.CollectionActor.Action;
 import nl.nn.adapterframework.collection.CollectionException;
+import nl.nn.adapterframework.collection.CollectorPipeBase.Action;
 import nl.nn.adapterframework.collection.ICollector;
 import nl.nn.adapterframework.configuration.ConfigurationException;
-import nl.nn.adapterframework.core.IForwardTarget;
 import nl.nn.adapterframework.core.PipeLineSession;
-import nl.nn.adapterframework.core.TimeoutException;
 import nl.nn.adapterframework.parameters.Parameter;
 import nl.nn.adapterframework.parameters.ParameterList;
-import nl.nn.adapterframework.parameters.ParameterValue;
 import nl.nn.adapterframework.parameters.ParameterValueList;
 import nl.nn.adapterframework.stream.Message;
-import nl.nn.adapterframework.stream.MessageOutputStream;
-import nl.nn.adapterframework.util.LogUtil;
-import nl.nn.adapterframework.util.StreamUtil;
-
-/**
- * Helper class to create Zip archives.
- *
- * @author  Gerrit van Brakel
- * @since   4.9.10
- */
-public class ZipWriter implements ICollector<IZipWritingElement> {
-	protected static Logger log = LogUtil.getLogger(ZipWriter.class);
-
-	public static final String PARAMETER_FILENAME="filename";
-	public static final String PARAMETER_CONTENTS="contents";
-
-	private @Getter ZipOutputStream zipoutput;
-	private boolean entryOpen=false;
-	private @Getter @Setter boolean closeOnExit;
+import nl.nn.adapterframework.stream.MessageContext;
+import nl.nn.adapterframework.stream.PathMessage;
+import nl.nn.adapterframework.util.FileUtils;
 
-	public ZipWriter(OutputStream resultStream, boolean closeOnExit) {
-		super();
-		this.closeOnExit=closeOnExit;
-		zipoutput=new ZipOutputStream(resultStream);
-	}
+public class ZipWriter implements ICollector<MessageZipEntry> {
 
-	public static ZipWriter getZipWriter(PipeLineSession session, String handlekey) {
-		return (ZipWriter)session.get(handlekey);
-	}
+	private static final MediaType MIMETYPE_ZIP = new MediaType("application", "zip");
+	static final String PARAMETER_FILENAME="filename";
+	static final String PARAMETER_CONTENTS="contents";
 
-	public static ZipWriter createZipWriter(PipeLineSession session, String handlekey, OutputStream resultStream, boolean closeOnExit) {
-		ZipWriter handle=new ZipWriter(resultStream,closeOnExit);
-		session.put(handlekey,handle);
-		handle.log.debug("opened new zipstream [{}]", handlekey);
-		return handle;
-	}
+	private final boolean includeFileHeaders;
 
-	public static void configure(Action action, ParameterList parameterList) throws ConfigurationException {
+	static void configure(Action action, ParameterList parameterList) throws ConfigurationException {
```

Comment:
- Why is this method static?
- Is `configure` the right name? Perhaps `validate` is better, since that's what it really  seems to do? Or `validateConfiguration`?